From 0ae2aa6a43a852cb0326faade1532c02aa7617a4 Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Fri, 20 Apr 2018 09:50:04 +0200
Subject: [PATCH 156/173] umount: add note about --lazy

Unfortunately, it's pretty common that users on production systems use
lazy umount to fix some FS issues. The usual result is unwanted system
reboot, because -l is not the right way how to fix unreachable NFS or
mess with local FS with submounts.

Note that after lazy umount /proc/self/mountinfo does not contain the
FS entry, but kernel still references the FS. It makes it very
difficult to debug.

Addresses: https://bugzilla.redhat.com/show_bug.cgi?id=1566674
Suggested-by:  Steve Dickson <steved@redhat.com>
Signed-off-by: Karel Zak <kzak@redhat.com>
Upstream: http://github.com/karelzak/util-linux/commit/031800ff6c66b4d62229cfd116195950a718a21c
---
 sys-utils/umount.8 | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/sys-utils/umount.8 b/sys-utils/umount.8
index f0a712b06..71c2155b4 100644
--- a/sys-utils/umount.8
+++ b/sys-utils/umount.8
@@ -54,7 +54,8 @@ working directory there, or when a swap file on it is in use.  The
 offending process could even be
 .B umount
 itself - it opens libc, and libc in its turn may open for example locale
-files.  A lazy unmount avoids this problem.
+files.  A lazy unmount avoids this problem, but it may introduce another
+issues. See \fB\-\-lazy\fR description bellow.
 .SH OPTIONS
 .TP
 \fB\-a\fR, \fB\-\-all\fR
@@ -107,6 +108,13 @@ Unmount without writing in
 Lazy unmount.  Detach the filesystem from the filesystem hierarchy now,
 and cleanup all references to the filesystem as soon as it is not busy
 anymore.  (Requires kernel 2.4.11 or later.)
+
+A system reboot would be expected in near future if you're going to use this
+option for network filesystem or local filesystem with submounts.  The
+recommended use-case for \fBumount -l\fR is to prevent hangs on shutdown due to
+an unreachable network share where a normal umount will hang due to a downed
+server or a network partition. Remounts of the share will not be possible.
+
 .TP
 \fB\-O\fR, \fB\-\-test\-opts\fR \fIoptions,list\fR
 Indicate that the actions should only be taken on file systems with the
-- 
2.14.4

